### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by rezg-dev**
Completed task 6 (add comments section placeholder is currently displaying for task 8).

Will work on styling further before starting task 10 (deleting)

Thanks

Reza

### CODE REVIEW

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Reza, I can see everything is working but I'm just going to suggest some changes before we move on. Please see my comments about your axios fetches being moved from the useEffects into their own separate files. Could you please do this and resubmit please? 🙂

File: fe-nc-games/src/components/CommentsList.jsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { useParams } from 'react-router-dom';
+import CommentCard from './CommentCard';
+
+function CommentsList() {
+  const [commentsList, setCommentsList] = useState([]);
+  const [isLoading, setLoading] = useState(true);
+  const { review_id } = useParams();
+
+  useEffect(() => {
+    axios
```
Comment (johnhutcheon) [inline fe-nc-games/src/components/CommentsList.jsx]: Can we move this axios to a separate file called api.js or utils.js please?

File: fe-nc-games/src/components/IndividualReviewPage.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import CommentsList from './CommentsList';
+
+function IndividualReviewPage() {
+  const { review_id } = useParams();
+
+  const [reviewDetails, setReviewDetails] = useState({});
+  const [isLoading, setLoading] = useState(true);
+
+  useEffect(() => {
+    axios
```
Comment (johnhutcheon) [inline fe-nc-games/src/components/IndividualReviewPage.jsx:14]: Can we move this axios to a separate file called api.js or utils.js please?

File: fe-nc-games/src/components/CommentsList.jsx (hunk 1 above)
Comment (rezg-dev) [reply fe-nc-games/src/components/CommentsList.jsx]: Sure, that should now be done...
added a utils folder and a file called apiRequests.

Should I put all the api requests in here and reference in the same way in other components?

Cheers,
Reza

File: fe-nc-games/src/ReviewDetails.jsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import IndividualReviewPage from "./components/IndividualReviewPage";
+
+
+function ReviewDetails () {
+    console.log('in the Review Details')
```
Comment (B7Bprog) [inline fe-nc-games/src/ReviewDetails.jsx]: Please remove all logs!

File: fe-nc-games/src/components/CommentsList.jsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  useEffect(() => {
+    getComments(review_id)
+    .then((response) => {
+      setCommentsList(response.data);
+      setLoading(false);
+    });
+  }, []);
+
+  if (isLoading) {
+    return <p>Loading Items...</p>;
```
Comment (B7Bprog) [inline fe-nc-games/src/components/CommentsList.jsx]: This p tag makes the text very small and hard to read / notice.

File: fe-nc-games/src/components/CommentsList.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  return (
+    <>
+      <section>
```
Comment (B7Bprog) [inline fe-nc-games/src/components/CommentsList.jsx]: When there are no comments, please indicate that to the user with a message like: 'No comments yet'.

File: fe-nc-games/src/components/CommentsList.jsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        {commentsList.map((comment) => {
+          return (
+            <CommentCard
```
Comment (B7Bprog) [inline fe-nc-games/src/components/CommentsList.jsx]: A simpler way is to pass 'comment' down as a prop.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Reza, Please see my comments, refactor your code, then its good to merge.

File: fe-nc-games/src/ReviewDetails.jsx (hunk 3 above)
Comment (rezg-dev) [reply fe-nc-games/src/ReviewDetails.jsx]: Done.

File: fe-nc-games/src/components/CommentsList.jsx (hunk 4 above)
Comment (rezg-dev) [reply fe-nc-games/src/components/CommentsList.jsx]: Changed to h2.

File: fe-nc-games/src/components/CommentsList.jsx (hunk 5 above)
Comment (rezg-dev) [reply fe-nc-games/src/components/CommentsList.jsx]: Added if statement:
  if (commentsList.length === 0) {
    return <p>No comments yet, why not add one?</p>
  }

File: fe-nc-games/src/components/CommentsList.jsx (hunk 6 above)
Comment (rezg-dev) [reply fe-nc-games/src/components/CommentsList.jsx]: Used spread operator to simplify.

**Pull request merged**